aya/src/sys/bpf.rs

```diff
@@ -382,7 +383,8 @@ pub(crate) fn bpf_link_create(
     target: LinkTarget<'_>,
     attach_type: bpf_attach_type,
     btf_id: Option<u32>,
-    flags: u32,
+    flags: Option<u32>,
```

Comment:
This is a little confusing to me `flags` and `ordering` are mutually exclusive, so it seems a bit odd to have them both as `Option` fields. What's the expected behaviour of `flags: Some(1), ordering: Some(2)`.
Wouldn't `flags: Option<u32>, expected_revision: Option<u32>` have less footguns?

Reply:
+1 I tripped myself up a bit here as well :/ I'll go with your suggestion!

Reply:
This was a bit confusing... `bpf_link_create` was pretty generic across program types until now when we added support for creating Mprog links.

I flattened it here, just not it tripped clippy's too many arguments checker so I had to add an allow here, but it's much more readable 👍